### ISSUE TITLE
Investigate removal of Deploy disable states

### ISSUE THREAD

**Issue by dotNomad** [posit-dev/publisher#2344]

**Comment by dotNomad** [comment]
Ultimately I split this out into 3 sub-issues to remove the disable states. The only one that can be removed quickly is #2345 and I will have a branch up for that in a moment.

The others are a bit more difficult. It is possible to get to a point where the only disable state we have is "a Deployment has not been chosen". I don't think we can get away from that one.

**Comment by jonkeane** [comment]
> The others are a bit more difficult. It is possible to get to a point where the only disable state we have is "a Deployment has not been chosen". I don't think we can get away from that one.

Agreed that no deployment has been chosen is a reasonable state to disable the button.

Disabling while deploying also might be reasonable to leave as is. Certainly the state(s) one can get into by pushing that button multiple times are funky. And we do have the cancel action which lets someone get out of the disabled state if they needed to.
